wallet/storage/src/internal/store_tx.rs

```diff
@@ -379,6 +379,141 @@ impl_read_ops!(StoreTxRw);
 impl_read_ops!(StoreTxRoUnlocked);
 impl_read_ops!(StoreTxRwUnlocked);
 
+impl<T> WalletStorageReadLocked for &mut T
```

Comment:
Those trait implementations will not be needed if we change the Signer trait to accept a &mut T instead.